step-automation-packages/step-automation-packages-manager/src/main/java/step/automation/packages/AutomationPackageServices.java

```diff
@@ -64,11 +70,31 @@ public void deleteAutomationPackage(@PathParam("name") String automationPackageN
     @Consumes(MediaType.MULTIPART_FORM_DATA)
     @Produces(MediaType.TEXT_PLAIN)
     @Secured(right = "automation-package-write")
-    public String createAutomationPackage(@FormDataParam("file") InputStream uploadedInputStream,
+    public String createAutomationPackage(@FormDataParam("file") InputStream automationPackageInputStream,
                                           @FormDataParam("file") FormDataContentDisposition fileDetail,
                                           @Context UriInfo uriInfo) throws Exception {
-        return automationPackageManager.createAutomationPackage(uploadedInputStream, fileDetail.getFileName(), getObjectEnricher());
+        return automationPackageManager.createAutomationPackage(automationPackageInputStream, fileDetail.getFileName(), getObjectEnricher());
+    }
 
+    // TODO: add new right to the permission matrix
+    @POST
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("/execute")
+    @Secured(right = "automation-package-execute")
+    public List<String> executeAutomationPackage(@FormDataParam("file") InputStream automationPackageInputStream,
+                                                 @FormDataParam("file") FormDataContentDisposition fileDetail,
+                                                 @Context UriInfo uriInfo) throws Exception {
+        // TODO: execution parameters map?
+        // TODO: add filters for plans
```

Comment:
The filter for plans could be step.core.execution.model.ExecutionParameters#artefactFilter